workloads/deployment/rbac.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: deployment-clusterrole-poc
+rules:
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - "*"
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: component-deployment-workload-poc
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: deployment-clusterrole-poc
+subjects:
+  - kind: ServiceAccount
+    name: crossplane
+    namespace: crossplane-system
```

Comment:
This assume user using `crossplane` and `crossplane-system` which can be changed in installation. We should point it out.